File: Others/medium/SearchInSortedMatrix.py
def searchInSortedMatrix(matrix, target):
    current_row = len(matrix) - 1
    current_col = 0

    while current_col >= 0 and current_col < len(matrix[0]) and current_row >= 0 and current_row < len(matrix):
        if matrix[current_row][current_col] == target:
            return [current_row, current_col]

        if matrix[current_row][current_col] > target:
            current_row -= 1
        else:
            current_col += 1

    return [-1, -1]

File: Golang/medium/search-in-sorted-matrix.go
package main

func SearchInSortedMatrix(matrix [][]int, target int) []int {
	rowStart := len(matrix) - 1
	colStart := 0

	for colStart < len(matrix[0]) && rowStart >= 0 {
		if matrix[rowStart][colStart] == target {
			return []int{rowStart, colStart}
		}
		if matrix[rowStart][colStart] > target {
			rowStart -= 1
		} else {
			colStart += 1
		}
	}

	return []int{-1, -1}
}
